frontend/src/components/Classroom/ClassroomList.vue

```diff
@@ -0,0 +1,134 @@
+<script setup>
+import { onMounted, ref, watch } from 'vue';
+import classroomService from '@/api/classrooms';
+import DeleteIcon from '@/assets/img/delete_icon.svg';
+import { parseDates } from '@/helpers/parse';
+
+const classrooms = ref(null);
+const searchTerm = ref('');
+const fields = {
+  name: 'Name',
+  capacity: 'Capacity',
+  createdAt: 'Created At',
+};
+
+onMounted(() => filterClassrooms(''));
+watch(searchTerm, newSearchTerm => filterClassrooms(newSearchTerm));
+const filterClassrooms = (search = '') => {
+  classroomService.getAllClassrooms().then(resp => {
+    const parsed = parseDates(resp);
+    const filtered = parsed.filter(room => room.name.toLowerCase().includes(search.toLowerCase()));
+    classrooms.value = filtered;
+  });
+};
```

Comment:
This code does two things (at least) that are not really necessary in my opinion.

The first is that it fetches the classrooms every time the user enters a search term. The search in this instance is client side so the fetching only serves to make sure the latest data is in the list before searching. But the page in itself does not make sure of this, only in this specific instance. If another user (or the same user in a different tab) adds or deletes a classroom the list is not updated. So there is inconsistency in the user experience. If the filtering were done on the backend then having the latest data is a side-effect of doing it there. So I would say either:
- move this filtering to the backend (does not need to be through the database but is still in memory),
- move the filtering into a computed and only fetch the classrooms on the mounted event. 

Generally, I would say to make the search on the backend, and do it through the database with an offset and limit. But since this is at the end of this phase of the project I would advise moving this into a computed and just fetch on mounted. By moving it to a computed, it removes the need for a watcher, for which I generally advise to avoide.

A note: I will have other feedback on the whole mounted event trigger in a separate comment that might affect this advise (if you choose to follow it that is).